oauth-common/src/main/java/io/strimzi/kafka/oauth/validator/JWTSignatureValidator.java

```diff
@@ -97,6 +107,20 @@ public JWTSignatureValidator(String keysEndpointUri,
         this.skipTypeCheck = skipTypeCheck;
         this.audience = audience;
 
+        if (enableBouncyCastleProvider && !bouncyInstalled.getAndSet(true)) {
+            int installedPosition = Security.insertProviderAt(new BouncyCastleProvider(), bouncyCastleProviderPosition);
+            log.info("BouncyCastle security provider installed at position: " + installedPosition);
+
+            if (log.isDebugEnabled()) {
+                StringBuilder sb = new StringBuilder("Installed security providers:\n");
+                for (Provider p: Security.getProviders()) {
+                    sb.append("  - " + p.toString() + "  [" + p.getClass().getName() + "]\n");
+                    sb.append("   " + p.getInfo() + "\n");
+                }
+                log.debug(sb.toString());
```

Comment:
this should log all the installed security providers but it happens only if the outer `if (enableBouncyCastleProvider && !bouncyInstalled.getAndSet(true))` is true. Is it really what we want? Shouldn't we log the installed security providers anyway?

Reply:
This is the way I intended. Usually you're not really interested in security providers, you're working with what's packaged and enabled in JDK. If you really want to see the providers for some reason, you can just enable bouncy castle provider in the config, and it will be added at the end, not overwriting any existing providers, but you'll also see them all listed.